### PR TITLE
Bug/decoding uri

### LINKED ISSUE #3164: [std] File server example can not handle url encoded characters
https://deno.land/std/http/file_server.ts can not handle urls like /test%20file.txt - it does not seem to parse URL encoded characters like %20.

### PR DESCRIPTION
**Pull request by AleksandrukTad**
Hey its my first pr to open source project, let me know if I have to do some additional stuff.
Was not able to test it with actual files, but now if you go to 0.0.0.0:4500/test test.txt fsPath will be with space not %20.

Fixes #3164

### CODE REVIEW

**Comment by AleksandrukTad** [comment]
Hey, I updated description remove ds_store. Unfortunately i do not know how to test this :/ Do you have like a crash course on the testing library that you are using?

**Comment by AleksandrukTad** [comment]
So my idea to do this test is:

Create file with the space for example "test test.txt",
run the server 
try to get the file go to localhost:port/test%20test.txt
expect status code 200.
Remove the file.

What do you think is this the right apporach?

**Comment by bartlomieju** [comment]
> So my idea to do this test is:
> 
> Create file with the space for example "test test.txt",
> run the server
> try to get the file go to localhost:port/test%20test.txt
> expect status code 200.
> Remove the file.
> 
> What do you think is this the right approach?

No need to create and remove the file, just put it into `std/http/testdata` directory and leave it there. Otherwise sounds good :+1:

**Comment by AleksandrukTad** [comment]
Hey I have done the unit test for this change, but since I pulled latest upstream is failing on sccache :/

**Comment by zekth** [comment]
> There still seems to be a //cli/.DS_Store file. Please remove this file.

We need to add this in the gitignore

File: std/http/file_server_test.ts [hunk 1]
```diff
@@ -83,3 +83,17 @@ test(async function serveFallback(): Promise<void> {
     killFileServer();
   }
 });
+
+test(async function serveFallback(): Promise<void> {
+  await startFileServer();
+  try {
+    const res = await fetch(
+      "http://localhost:4500/std/http/testdata/test%20file.txt"
```
Comment (zhmushan) [inline std/http/file_server_test.ts]: should be changed to `http://localhost:4500/http/testdata/test%20file.txt`

**Comment by AleksandrukTad** [comment]
@bartlomieju It seems like some conflicts have to be resolved. Can you please do it whenever you have a moment? Cheers

**Comment by nayeemrmn** [comment]
@AleksandrukTad You forgot to run `git submodule update` before your last commit. You'll need to `cd` into those submodules and checkout the commit that master has them on and then commit that.

**Comment by AleksandrukTad** [comment]
@nayeemrmn git submodule does not do anything if I cd into "third_party" :/.

**Comment by nayeemrmn** [comment]
> @nayeemrmn git submodule does not do anything if I cd into "third_party" :/.

That's not what I said to do 😅. Okay try this instead: `git reset HEAD~`, `git  submodule update` and then redo your last commit.

**Comment by zhmushan** [comment]
Following may be useful to you:
```
git remote add upstream https://github.com/denoland/deno
git fetch upstream
git rebase upstream/master
git push origin master -f
```

**Comment by AleksandrukTad** [comment]
@nayeemrmn Sorry I must have missundertood then. My git knowledge is pretty basic add/commit/push/pull/rebase. You solution worked thanks!

**Comment by AleksandrukTad** [comment]
@nayeemrmn @zhmushan any idea why this is failing tests only on windows?

**Comment by lucacasonato** [comment]
@AleksandrukTad Just push anything to the branch to return the test. @ry the plugin test seems to be flakey. Had the same issue in my last PR.

**Pull request merged**